tests/test_utils.py

```diff
@@ -125,6 +126,53 @@ def test_get_needed_absolute_time_range(m_dt):
 
 
 @freeze_time("2016-01-01 20:00:00", tz_offset=-8)
+def test_get_chunked_time_ranges_last_15m():
+    kairos_timing = {'start_relative': {'unit': 'minutes', 'value': '15'}}
+    now = datetime.datetime.now()
+    then = now - datetime.timedelta(minutes=15)
+    assert get_chunked_time_ranges({'chunking': {}}, kairos_timing) == [(then, now)]
+
+
+@freeze_time("2016-01-01 20:00:00", tz_offset=-8)
+def test_get_chunked_time_ranges_last_2h():
+    kairos_timing = {'start_relative': {'unit': 'hours', 'value': '2'}}
+    now = datetime.datetime.now()
+    results = get_chunked_time_ranges({'chunking': {}}, kairos_timing)
+    assert len(results) == 4
+    for i in xrange(len(results)):
+        offset = datetime.timedelta(minutes=(i * 30))
+        assert results[i][1] == now - offset
+        assert results[i][0] == now - offset - datetime.timedelta(minutes=30)
+
+
+@freeze_time("2016-01-01 20:00:00", tz_offset=-8)
+def test_get_chunked_time_ranges_last_12h():
+    kairos_timing = {'start_relative': {'unit': 'hours', 'value': '12'}}
+    now = datetime.datetime.now()
+    results = get_chunked_time_ranges({'chunking': {}}, kairos_timing)
+    assert len(results) == 6
+    for i in xrange(len(results)):
+        offset = datetime.timedelta(minutes=(i * 120))
+        assert results[i][1] == now - offset
+        assert results[i][0] == now - offset - datetime.timedelta(minutes=120)
+
+
+@freeze_time("2016-01-01 20:00:00", tz_offset=-8)
+def test_get_chunked_time_ranges_last_5h15m():
```

Comment:
nit but shouldn't this be "...ranges_last_2h15m" since the time is relative 135minutes?